src/App.tsx

```diff
@@ -1,14 +1,55 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [selectedTodos, setSelectedTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [todoSelected, setTodoSelected] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [status, setStatus] = useState('all');
+
+  useEffect(() => {
+    getTodos()
+      .then(todos => {
+        setSelectedTodos(todos);
+        setIsLoading(false);
+      });
+  }, []);
+
+  const initialFilter = () => {
+    switch (status) {
+      case 'active':
+        return selectedTodos.filter(todo => !todo.completed);
+
+      case 'completed':
+        return selectedTodos.filter(todo => todo.completed);
+
+      case 'all':
+      default:
+        return selectedTodos;
+    }
+  };
+
+  const todoFilter = () => {
+    if (query === '') {
+      return initialFilter();
+    }
+
+    return initialFilter()
+      .filter(todo => todo.title.toLowerCase().includes(query.toLowerCase()));
+  };
+
+  const filteredTodos = todoFilter();
```

Comment:
Would be good to wrap this value to the useMemo, to prevent memory leaks. And wrap the functions above into the useCallback() if you will use them in the dependency arrays